### PR TITLE
Add a Security Group for Windows AMI Creation

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This PR adds a security group to use for the manual portion of Windows AMI creation. It allows RDP access and has a name to make it easy to reference when spinning up an instance with a VM import AMI.

## 💭 Motivation and context ##

Access is needed to finish the setup for Windows AMIs after the initial import from VM. This provides a static security group to use in a convenient way since this process is currently manual.

## 🧪 Testing ##

Automated tests pass. I have applied these changes to the Staging environment and was able to use this Security Group to access an instance for configuration.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: images/windows_ami_sg.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+# Allow ingress to instances via RDP (UDP)
+resource "aws_security_group_rule" "windows_ami_ingress_via_rdp_over_udp" {
+  security_group_id = aws_security_group.windows_ami.id
+  type              = "ingress"
+  protocol          = "udp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  from_port         = 3389
+  to_port           = 3389
+}
```
Comment (jsf9k) [inline images/windows_ami_sg.tf]:
```suggestion
# Allow ingress to instances via RDP
resource "aws_security_group_rule" "windows_ami_ingress_via_rdp" {
  for_each = toset(["tcp", "udp"])

  security_group_id = aws_security_group.windows_ami.id
  type              = "ingress"
  protocol          = each.value
  cidr_blocks       = ["0.0.0.0/0"]
  from_port         = 3389
  to_port           = 3389
}
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
I made one suggestion to avoid duplicating some code.

**Review by jsf9k — APPROVED** [review]

File: images/variables.tf [hunk 2]
```diff
@@ -141,3 +141,9 @@ variable "third_party_bucket_name_prefix" {
   description = "The prefix to use to name the S3 bucket for storing third-party files.  The bucket will be named with this prefix plus the account type (e.g. production or staging)."
   default     = "cisa-cool-third-party"
 }
+
+variable "windows_ami_sg_name" {
+  type        = string
+  description = "The name to associate with the Security Group that allows access for finalizing Windows AMI configuration."
```
Comment (dav3r) [inline images/variables.tf]: I don't think we capitalize "security group" - AWS doesn't do that in their docs:
```suggestion
  description = "The name to associate with the security group that allows access for finalizing Windows AMI configuration."
```

File: images/windows_ami_sg.tf [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Security group for manually finishing AMI setup for imported Windows AMIs
+resource "aws_security_group" "windows_ami" {
+  # We include a name for convenience because this Security Group will be
```
Comment (dav3r) [inline images/windows_ami_sg.tf]: Unnecessary capitalization:
```suggestion
  # We include a name for convenience because this security group will be
```

**Review by dav3r — APPROVED** [review]
👍  aside from some capitalization foo that I mentioned.

File: images/variables.tf (hunk 2 above)
Comment (mcdonnnj) [reply images/variables.tf]: Please see 97ee8a1 (I did it manually to update the README at the same time).

File: images/windows_ami_sg.tf (hunk 3 above)
Comment (mcdonnnj) [reply images/windows_ami_sg.tf]: Please see 97ee8a1.

**Pull request merged**